### PR TITLE
CRM-19610 : Search preferences changes are not updated

### PR DESCRIPTION
**Pull request by seamuslee001**
* [CRM-19610: Fatal when creating InnoDB fts indexes](https://issues.civicrm.org/jira/browse/CRM-19610)

### CODE REVIEW

**Comment by seamuslee001** [comment]
@jitendrapurohit i have made some changes to your PR and rebased it, I think the tests will pass now, Can you check to see if it makes sense to you?

File: Civi/Core/SettingsBag.php [hunk 1]
```diff
@@ -352,7 +352,7 @@ protected function setDb($name, $value) {
     }
     $dao->find(TRUE);
 
-    if (isset($metadata['on_change'])) {
+    if (isset($metadata['on_change']) && !($value === 0 && ($dao->value === NULL || unserialize($dao->value) === 0))) {
```
Comment (jitendrapurohit) [inline Civi/Core/SettingsBag.php]: Thanks @seamuslee001. It passes the test now but the main issue seems to be appearing after this change. As [$value is supposed to be of type mixed](https://github.com/civicrm/civicrm-core/blob/master/Civi/Core/SettingsBag.php#L325) in this function, `===` will always fail for `'0'` if passed as a string.

**Comment by seamuslee001** [comment]
@jitendrapurohit i have altered the check to allow for possibility of '0' and locally tests are still passing, I note that your most recent change still fails the tests. I'm just not 100% sure if my logic is correct and don't know why mine would pass but yours wouldn't

**Review by jitendrapurohit — APPROVED** [review]
Me neither, but i can confirm that this change fixes the original issue. So, i think this could be merged as it also passes the test build.

**Comment by seamuslee001** [comment]
ping @eileenmcnaughton @totten

**Comment by eileenmcnaughton** [comment]
This seems fine to me - note there is an open critical issue that would be probably fixed by this too. I think @nganivet logged it some time back?

**Comment by eileenmcnaughton** [comment]
Ah - linked JIRA issue is wrong?

**Comment by seamuslee001** [comment]
@eileenmcnaughton nope correct i believe its the same as Jitendra's and https://github.com/civicrm/civicrm-core/pull/10453 should be merged as well

**Pull request merged**